### PR TITLE
Gutenboarding: Some more header/style adjustments

### PR DESCRIPTION
**Pull request by chriskmnds**
#### Changes proposed in this Pull Request

Some more header cleanup, also matches the figma a little better.

**before**

Screenshot 2020-03-30 at 11 16 44 AM

**after**

Screenshot 2020-03-30 at 11 15 54 AM

#### Testing instructions

http://calypso.localhost:3000/gutenboarding

### CODE REVIEW

File: client/landing/gutenboarding/components/header/style.scss [hunk 1]
```diff
@@ -58,22 +58,6 @@ $padding--gutenboarding__header-domain-picker-button: 0.3em 0.5em;
...
-		.gutenboarding__site-title {
-			color: black;
-		}
-	}
-
-	.gutenboarding__site-title {
-		padding: 0;
-	}
-
-	> span {
```
Comment (chriskmnds) [inline client/landing/gutenboarding/components/header/style.scss]: Need to double check if this is any longer needed.

File: client/landing/gutenboarding/components/header/style.scss [hunk 2]
```diff
@@ -58,22 +58,6 @@ $padding--gutenboarding__header-domain-picker-button: 0.3em 0.5em;
...
-		display: flex;
-		flex-direction: column;
-		align-items: flex-start;
-	}
-	> .dashicon {
```
Comment (chriskmnds) [inline client/landing/gutenboarding/components/header/style.scss]: Same - need to double check if any longer needed.

File: client/landing/gutenboarding/components/header/style.scss (hunk 1 above)
Comment (alshakero) [reply client/landing/gutenboarding/components/header/style.scss]: The `span` is used here: https://github.com/Automattic/wp-calypso/blob/update/gutenboarding-header-styles/client/landing/gutenboarding/components/domain-picker-button/index.tsx#L79

But I can't tell what's the harm of removing it.

File: client/landing/gutenboarding/components/header/style.scss (hunk 2 above)
Comment (alshakero) [reply client/landing/gutenboarding/components/header/style.scss]: The icon is also used here (assuming `DashIcon` component uses it internally): https://github.com/Automattic/wp-calypso/blob/update/gutenboarding-header-styles/client/landing/gutenboarding/components/domain-picker-button/index.tsx#L80

File: client/landing/gutenboarding/components/header/style.scss (hunk 1 above)
Comment (chriskmnds) [reply client/landing/gutenboarding/components/header/style.scss]: I can't tell either. Let's just remove.

**Comment by chriskmnds** [comment]
Thanks for taking the time @alshakero ! :-)

**Pull request merged**